### PR TITLE
add documentation for multi panel

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

File: en/developer_guide/multiPanel.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+.. code-block:: yml
+    parameters:
+        open_orchestra_workflow_function_admin.subscriber.add_workflow_link.class: OpenOrchestra\WorkflowFunctionAdminBundle\EventSubscriber\AddWorkFlowLinkSubscriber
+    services:
+        open_orchestra_workflow_function_admin.subscriber.add_workflow_link:
+            class: %open_orchestra_workflow_function_admin.subscriber.add_workflow_link.class%
+            arguments:
+                - @router
+            tags:
+                - { name: kernel.event_subscriber }
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: The code block is not displayed in "view" mode

File: en/developer_guide/multiPanel.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+-----
+
+Panel System
+~~~~~~~~~~~~
+
+To add a new panel, you simply have to had a ``_self_panel_`` link to the facade of the User page.
+
+.. code-block:: php
+   $linkName =
+   $facade->addLink("_self_panel_" + $position + $paneltitle, $route);
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: The code block is not displayed in "view" mode

File: en/developer_guide/multiPanel.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+         * @param UserFacadeEvent $event
+         */
+        public function postUserTransformation(UserFacadeEvent $event)
+        {
+            $facade = $event->getUserFacade();
+            $facade->addLink('_self_panel_2_workflow_right',
+                $this->router->generate($workflowRouteName,
+                    array('userId' => $facade->id),
+                    UrlGeneratorInterface::ABSOLUTE_URL)
+                    );
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: incorrect psr2 formatting

File: en/developer_guide/multiPanel.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Open Orchestra implement a panel system to make this work easier.
+Here is an example of how we use this panel system to add a workflow panel to Users page.
+
+Usage
+-----
+
+Panel System
+~~~~~~~~~~~~
+
+To add a new panel, you simply have to had a ``_self_panel_`` link to the facade of the User page.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: Not clear, I understand that this is specific to the user page, but I guess it's more than that. Maybe also explain which facade we are talking about (I guess UserCollection ?)

File: en/developer_guide/multiPanel.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
+Change an administration view to multi panel
+============================================
+
+Context
+-------
+
+When working with administration choice, sometimes you need to add your own forms or contents to the existing one.
+Open Orchestra implement a panel system to make this work easier.
+Here is an example of how we use this panel system to add a workflow panel to Users page.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: I don't understand what this feature will bring. Is it about the display, the code, or both ?

File: en/developer_guide/multiPanel.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
+Add a panel to an administration view
+=====================================
+
+Context
+-------
+
+When working with administration back office, sometimes you need to add your own forms or contents to the existing one.
+Open Orchestra implement a panel system to make this work easier.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: implements

File: en/developer_guide/multiPanel.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+Here is an example of how we use this panel system to add a workflow panel to Users page.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: I don't understand what this feature will bring. Is it about the display, the code, or both ?

File: en/developer_guide/multiPanel.rst [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+Context
+-------
+
+When working with administration back office, sometimes you need to add your own forms or contents to the existing one.
+Open Orchestra implements a system to easily add form to your administration page in a new panel.
+
+Panel System
+~~~~~~~~~~~~
+
+To add a new panel, you have to had a ``_self_panel`` link to the facade of the page.
```
Comment (nicolasThal) [inline en/developer_guide/multiPanel.rst]: To display a new panel, add a link to the facade, prefixed by `_self_panel`

File: en/developer_guide/multiPanel.rst [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+.. code-block:: php
+
+    $facade->addLink("_self_panel_" + $position + "_" + $panelTitle,
```
Comment (nicolasThal) [inline en/developer_guide/multiPanel.rst]: do not split the line

File: en/developer_guide/multiPanel.rst [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        $route);
+
+The link name is a key to choose position and name of your panel:
```
Comment (nicolasThal) [inline en/developer_guide/multiPanel.rst]: What is the link name?

You should either : 
- introduce this concept
- say what it refers to

File: en/developer_guide/multiPanel.rst [hunk 11]
```diff
@@ -0,0 +1,83 @@
+Add panel to an administration view
+===================================
+
+Context
+-------
+
+When working with administration back office, sometimes you need to add your own forms or contents to the existing one.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: ones

File: en/developer_guide/multiPanel.rst [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+Open Orchestra implements a system to easily add form to your administration page in a new panel.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: "forms" or "a form"

File: en/developer_guide/multiPanel.rst [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+Panel System
+~~~~~~~~~~~~
+
+Each administration facade have a list of links used by ajax to load their form.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: "facades" or "has"

File: en/developer_guide/multiPanel.rst [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+To display a new panel, add a link to the facade, prefixed by ``_self_panel``.
+
+.. code-block:: php
+
+    $linkName = "_self_panel_" + $position + "_" + $panelTitle;
+    $facade->addLink($linkName, $route);
+
+The link name is a key to find your link in an array.
+Open Orchestra also use this key to choose position and name of your panel:
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: uses
the position
"an array" => what array ? If it's useful to talk about it it should be clear what is meant

File: en/developer_guide/multiPanel.rst [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+
+    - Add a number after "_self_panel" prefixed by "_" to choose the position of your panel (begin with 1 because ``_self_form`` link will always be in position 0)
+    - Add the title of your panel after the position prefixed by "_" to choose the menu name
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: This bulleted list displays greyed on the view mode, not sure it's on purpose.

File: en/developer_guide/multiPanel.rst [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+Example
+~~~~~~~
+
+Here is an example of how we use panel system to add a workflow form to Users page.
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: the panel system

File: en/developer_guide/multiPanel.rst [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+.. image:: ../../images/user_panel_form.png
+
+To add this link, you can use an event subscriber on postUserTransformation :
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: the UserFacadeEvents::POST_USER_TRANSFORMATION event

File: en/developer_guide/multiPanel.rst [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+         */
+        public static function getSubscribedEvents()
+        {
+            return array(
+                UserFacadeEvents::POST_USER_TRANSFORMATION => 'postUserTransformation',
+            );
+        }
+    }
+
+and register the event subscriber as service in a configuration file :
```
Comment (juchi) [inline en/developer_guide/multiPanel.rst]: as a service

**Comment by juchi** [comment]
It's still very unclear what a panel is without the help of the image. Speaking of "tabs" somewhere would be much better.

File: en/developer_guide/multiPanel.rst [hunk 19]
```diff
@@ -0,0 +1,83 @@
+Add panel to an administration view
+===================================
+
+Context
+-------
+
+When working with administration back office, sometimes you need to add your own forms or contents to the existing ones.
```
Comment (ngilain) [inline en/developer_guide/multiPanel.rst]: I find this introduction a little confusing. I don't see the aim of the doc. Maybe you could rephrase it.

File: en/developer_guide/multiPanel.rst [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+The link name is a key to find the link later.
+Open Orchestra also uses this key to configure your panel:
+
+- Add a number after "_self_panel" prefixed by "_" to choose the position of your tab (begin with 1 because ``_self_form`` link will always be in position 0)
+- Add the title of your panel after the position prefixed by "_" to choose the tab name
+
+Example
+~~~~~~~
+
+Here is an example of how we use the panel system to add a workflow form to Users page.
```
Comment (ngilain) [inline en/developer_guide/multiPanel.rst]: [...] of how is used the panel [...]

Comment (juchi) [inline en/developer_guide/multiPanel.rst]: "how the panel is used" would be more correct.

File: en/developer_guide/multiPanel.rst [hunk 21]
```diff
@@ -0,0 +1,83 @@
+Add panel to an administration view
+===================================
+
+Context
+-------
+
+Sometimes you need to add your own forms to the existing ones in the Administration Back office to add some options to the administration items.
```
Comment (nicolasThal) [inline en/developer_guide/multiPanel.rst]: Can you add a line return

**Comment by nicolasThal** [comment]
@itkg-ngilain @juchi @itkg-rboessel I have update the context part

File: en/developer_guide/multi_panel.rst [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+Context
+-------
+
+In Open Orchestra, if you want to edit an element, you can only display the full form. In some cases
+you want to embillish the edition by splitting the full form in some smaller one.
+
+Open Orchestra already provides you a way to `use a custom Backbone view`_.
+This solution may be possible but will require a large customization of the form view.
+
+To prevent you from doing this work, Open Orchestra provide a way to add multiple panel in your edition form.
```
Comment (ngilain) [inline en/developer_guide/multi_panel.rst]: Open Orchestra provides
to add multiple tabs in

File: en/developer_guide/multi_panel.rst [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+
+Panel System
+~~~~~~~~~~~~
+
+Each element facade contains a list of links that may be used to display more specific information about them.
```
Comment (ngilain) [inline en/developer_guide/multi_panel.rst]: I think something lacks here : behind links there are some controllers dedicated to the render of the smaller parts of the main form.
So I think we should start the explanation by the conception of this subform controllers, then tell how to use it.

File: en/developer_guide/multi_panel.rst [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+Open Orchestra also uses this key to configure your panel:
+
+- Add a number after "_self_panel" prefixed by "_" to choose the position of your tab (begin with 1 because
+ the ``_self_form`` link will always be in position 0)
+- Add the title of your panel after the position prefixed by "_" to choose the tab name
+
+Example
+~~~~~~~
+
+Here is an example of how the panel is used to add a workflow form to Users page.
```
Comment (ngilain) [inline en/developer_guide/multi_panel.rst:47]: This snapshot is good, but I think it should be used on the beggining of the doc, to clearly show what it's talking about.

**Comment by ngilain** [comment]
The doc is now clearer than on the first version, but a lot of components are involved in this process (facade, backbone view, subform controllers, subscriber, etc ...)
Maybe a complete diagram of the workflow involved here could help to have a global look of the process

File: en/developer_guide/multi_panel.rst [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+.. image:: ../../images/user_panel_form.png
+
+Panel System
+~~~~~~~~~~~~
+
+Each element facade contains a list of meta links that may be used to display specifics meta-informations
+or forms to add, modify or delete them.
+Most commons links are :
+
+- ``_self_add`` to add element
```
Comment (ngilain) [inline en/developer_guide/multi_panel.rst]: Maybe something more complete like 'url to the add form action'
This comment apply to all others links

File: en/developer_guide/multi_panel.rst [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+Edit page of administration items use ``_self_form`` link to display the edit form.
+
+To add a new panel with your own form to this one, add a link to the facade, prefixed by ``_self_panel``.
+
+.. code-block:: php
+
+    $linkName = "_self_panel_" + $position + "_" + $panelTitle;
+    $facade->addLink($linkName, $route);
+
+The link name is a key to find the link later.
```
Comment (ngilain) [inline en/developer_guide/multi_panel.rst]: What is behind 'later' ?

File: en/developer_guide/multi_panel.rst [hunk 27]
```diff
@@ -0,0 +1,105 @@
...
+~~~~~~~~~~~~
+
+Each element facade contains a list of meta links that may be used to display specifics meta-informations
+or forms to add, modify or delete them.
+Most commons links are :
+
+- ``_self_add`` to add element
+- ``_self_form`` to edit element
+- ``_self_delete`` to delete element
+_ ``_self_meta`` to display meta-information
```
Comment (ngilain) [inline en/developer_guide/multi_panel.rst]: There's a rendering problem with the _self_meta point

**Comment by nicolasThal** [comment]
:+1:

File: en/developer_guide/multi_panel.rst [hunk 28]
```diff
@@ -0,0 +1,104 @@
+Add panel to an administration view
+===================================
+
+Context
+-------
+
+In Open Orchestra, if you want to edit an element, you can only display the full form. In some cases
+you want to embillish the edition by splitting the full form in some smaller one.
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: embellish ?
splitting something **into** something else

File: en/developer_guide/multi_panel.rst [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+
+Open Orchestra already provides you a way to `use a custom Backbone view`_.
+This solution may be possible but will require a large customization of the form view.
+
+To prevent you from doing this work, Open Orchestra provides a way to add multiple tab in your edition form like in the user edit page.
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: multiple tabs

File: en/developer_guide/multi_panel.rst [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+
+.. image:: ../../images/user_panel_form.png
+
+Panel System
+~~~~~~~~~~~~
+
+Each element facade contains a list of meta links that may be used to display specifics meta-informations
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: information is invariable : specific meta-information

File: en/developer_guide/multi_panel.rst [hunk 31]
```diff
@@ -0,0 +1,104 @@
...
+or forms to add, modify or delete them.
+Most commons links are :
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: The most common

File: en/developer_guide/multi_panel.rst [hunk 32]
```diff
@@ -0,0 +1,104 @@
...
+
+- ``_self_add`` url to the add form action
+- ``_self_form`` url to the edit form action
+- ``_self_delete`` url to the delete form action
+- ``_self_meta`` url to meta-information
+
+All of this links are used by open orchestra javascript to send ajax call and display differents administration forms.
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: these links
Open Orchestra's JavaScript code
ajax calls
different

File: en/developer_guide/multi_panel.rst [hunk 33]
```diff
@@ -0,0 +1,104 @@
...
+Edit page of administration items use ``_self_form`` link to display the edit form.
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: The edit page
uses

File: en/developer_guide/multi_panel.rst [hunk 34]
```diff
@@ -0,0 +1,104 @@
...
+
+- Add a number after "_self_panel" prefixed by "_" to choose the position of your tab (begin with 1 because the ``_self_form`` link will always be in position 0)
+- Add the title of your panel after the position prefixed by "_" to choose the tab name
+
+Example
+~~~~~~~
+
+Here is an example of how the panel is used to add a workflow form to Users page.
+
+To add this link without modifying the ``UserTransformer`` which create the facade, we recommand to dispatch an event subscriber
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: creates
recommend
an event is dispatched, a subscriber listen to events => end of the sentence to rephrase

File: en/developer_guide/multi_panel.rst [hunk 35]
```diff
@@ -0,0 +1,104 @@
...
+like ``UserFacadeEvents::POST_USER_TRANSFORMATION`` to modifying the facade by your own after the ``UserTransformer`` work:
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst]: to modifiy

File: en/developer_guide/multi_panel.rst [hunk 36]
```diff
@@ -0,0 +1,104 @@
...
+        /**
+         * @return array The event names to listen to
+         */
+        public static function getSubscribedEvents()
+        {
+            return array(
+                UserFacadeEvents::POST_USER_TRANSFORMATION => 'postUserTransformation',
+            );
+        }
+    }
```
Comment (juchi) [inline en/developer_guide/multi_panel.rst:79]: the PHP-doc is often dismissed in documentation for the sake of brevity

**Comment by veolys-nanne** [comment]
:+1:

**Pull request merged**